### PR TITLE
mac80211/hostapd: Fix setting radio htmode when using mesh mode

### PR DESCRIPTION
**Pull request by xsokolikx**
When configuring the radio in legacy mode from luci, the htmode is not set correctly to NOHT, causing the radio in mesh mode to be set to HT40. 
Also, disable_vht parameter needs to be set when using wpa_supplicant NOHT/HT20/HT40 modes.

### CODE REVIEW

File: package/network/services/hostapd/files/hostapd.sh [hunk 1]
```diff
@@ -765,6 +765,7 @@ wpa_supplicant_set_fixed_freq() {
 		NOHT) append network_data "disable_ht=1" "$N$T";;
 		HT20|VHT20) append network_data "disable_ht40=1" "$N$T";;
 		HT40*|VHT40*|VHT80*|VHT160*) append network_data "ht40=1" "$N$T";;
+		NOHT*|HT*) append network_data "disable_vht=1" "$N$T";;
```
Comment (adrianschmutzler) [inline package/network/services/hostapd/files/hostapd.sh:768]: Note that this is a case, so NOHT, HT20 and HT40* won't fall into this option ...

**Comment by adrianschmutzler** [comment]
Changes for different packages should be separate commits if possible. Despite, you lack a Signed-off-by in your commit message.

**Comment by xsokolikx** [comment]
Ok, i realized the same problem and was fixing it. Also adding sign-off.

**Comment by xsokolikx** [comment]
Commit splitted in two:

- hostapd: Add disable_vht when using NOHT/HT* modes

- mac80211: Fix setting radio htmode when using mesh mode

Thanks for the feedback.

File: package/network/services/hostapd/files/hostapd.sh (hunk 1 above)
Comment (adrianschmutzler) [reply package/network/services/hostapd/files/hostapd.sh:770]: You could actually merge this case with the one following it.

**Comment by adrianschmutzler** [comment]
hostapd commit lacks a description (about why it is necessary).

File: package/network/services/hostapd/files/hostapd.sh [hunk 2]
```diff
@@ -766,6 +766,9 @@ wpa_supplicant_set_fixed_freq() {
 		HT20|VHT20) append network_data "disable_ht40=1" "$N$T";;
 		HT40*|VHT40*|VHT80*|VHT160*) append network_data "ht40=1" "$N$T";;
 	esac
+	case "$htmode" in
+		NOHT*|HT*) append network_data "disable_vht=1" "$N$T";;
```
Comment (adrianschmutzler) [inline package/network/services/hostapd/files/hostapd.sh:770]: You could actually merge this case with the one following it.

**Comment by xsokolikx** [comment]
Added description to hostapd commit. Also, the hostapd commit has been merged with the next switch case.

**Comment by jow-** [comment]
Are we sure that `disable_vht` is recognized by all hostapd variants? Iirc 11n support used to be optional at some point which caused hostapd to not recognize certain options.

**Comment by xsokolikx** [comment]
The hostapd commit only affects wpa_supplicant configuration. After reviewing code from wpa-supplicant and wpa-supplicant-mini variants, disable_vht related parts exists on both:

```
#ifdef CONFIG_VHT_OVERRIDES
        if (ssid->disable_vht) {
                freq->vht_enabled = 0;
                return;
        }
#endif /* CONFIG_VHT_OVERRIDES */
```

**Comment by jow-** [comment]
And `CONFIG_VHT_OVERRIDES` ends up being set in all cases? Iirc there's some conditional compilation going on, depending on which wireless drivers were selected at hostapd compilation time.

**Comment by xsokolikx** [comment]
Yes, `CONFIG_VHT_OVERRIDES` is set in all cases:
```
~/package/network/services/hostapd$ grep -r CONFIG_VHT_OVERRIDES
files/wpa_supplicant-p2p.config:CONFIG_VHT_OVERRIDES=y
files/wpa_supplicant-mini.config:CONFIG_VHT_OVERRIDES=y
files/wpa_supplicant-full.config:CONFIG_VHT_OVERRIDES=y
files/wpa_supplicant-basic.config:CONFIG_VHT_OVERRIDES=y
```
Also, note that if _disable_vht_ parameter is not set, if you configure a radio using mesh on HT, it is set by default to VHT.

File: package/network/services/hostapd/files/hostapd.sh (hunk 2 above)
Comment (dangowrt) [reply package/network/services/hostapd/files/hostapd.sh:770]: Yes, you can use the default of the statement below on line 776:
```
*) append network_data "max_oper_chwidth=0" "$N$T";;
```
and turn that into
```
VHT20|VHT40) append network_data "max_oper_chwidth=0" "$N$T";;
*) append network_data "disable_vht=1" "$N$T";;
```

**Review by dangowrt — CHANGES_REQUESTED** [review]
Please merge your new `case` statement with the existing one below, as shown in the comment.
Apart from that lgtm, let's get it ready and merge.

**Review by dangowrt — APPROVED** [review]

**Comment by dangowrt** [comment]
cherry-picked onto master. thanks!

**Pull request closed without merging**